velox/exec/tests/SortBufferTest.cpp

```diff
@@ -0,0 +1,356 @@
+
+  for (const auto& testData : testSettings) {
+    auto sortBuffer = new SortBuffer(
+        testData.inputType,
+        testData.sortColumnIndices,
+        testData.sortCompareFlags,
+        1000,
+        pool_.get(),
+        &nonReclaimableSection_);
+
+    const std::shared_ptr<memory::MemoryPool> fuzzerPool =
+        memory::addDefaultLeafMemoryPool("VectorFuzzer");
+
+    std::vector<RowVectorPtr> inputVectors;
+    inputVectors.reserve(3);
+    for (size_t inputRows : {1000, 1000, 1000}) {
+      VectorFuzzer fuzzer({.vectorSize = inputRows}, fuzzerPool.get());
+      RowVectorPtr input = fuzzer.fuzzRow(inputType_);
+      sortBuffer->addInput(input);
+      inputVectors.push_back(input);
+    }
+    auto spillStats = sortBuffer->noMoreInput();
+
+    int expectedResultIndex = 0;
+    RowVectorPtr expectedResult = buildExpectedSortResult(inputVectors);
+    RowVectorPtr output = sortBuffer->getOutput();
+    while (output != nullptr) {
+      verifyData(output, expectedResult, expectedResultIndex);
+      output = sortBuffer->getOutput();
+    }
+  }
+}
+
+TEST_F(SortBufferTest, batchOutput) {
+  struct {
+    std::vector<size_t> numInputRows;
+    size_t outputBatchSize;
+    std::vector<size_t> expectedOutputBufferSizes;
+
+    std::string debugString() const {
+      return fmt::format(
+          "numInputRows:{}, outputBatchSize:{}, expectedOutputBufferSizes:{}, triggerSpill: {}",
+          std::string(numInputRows.begin(), numInputRows.end()),
+          outputBatchSize,
+          std::string(
+              expectedOutputBufferSizes.begin(),
+              expectedOutputBufferSizes.end()));
+    }
+  } testSettings[] = {
+      {{2000}, 10000, {2000}},
+      {{2000}, 2000, {2000}},
+      {{1024}, 1000, {1000, 24}},
+      {{1024, 1024, 1024}, 1000, {1000, 1000, 1000, 72}}};
+
+  for (const auto& testData : testSettings) {
+    auto sortBuffer = new SortBuffer(
+        inputType_,
+        sortColumnIndices_,
+        sortCompareFlags_,
+        testData.outputBatchSize,
+        pool_.get(),
+        &nonReclaimableSection_);
+
+    const std::shared_ptr<memory::MemoryPool> fuzzerPool =
+        memory::addDefaultLeafMemoryPool("VectorFuzzer");
+
+    std::vector<RowVectorPtr> inputVectors;
+    inputVectors.reserve(testData.numInputRows.size());
+    for (size_t inputRows : testData.numInputRows) {
+      VectorFuzzer fuzzer({.vectorSize = inputRows}, fuzzerPool.get());
+      RowVectorPtr input = fuzzer.fuzzRow(inputType_);
+      sortBuffer->addInput(input);
+      inputVectors.push_back(input);
+    }
+    auto spillStats = sortBuffer->noMoreInput();
+
+    int expectedOutputBufferIndex = 0;
+    RowVectorPtr output = sortBuffer->getOutput();
+    while (output != nullptr) {
+      ASSERT_EQ(
+          output->size(),
+          testData.expectedOutputBufferSizes[expectedOutputBufferIndex++]);
+      output = sortBuffer->getOutput();
+    }
+  }
+}
+
+TEST_F(SortBufferTest, spill) {
```

Comment:
Can you verify the spilling stats by following similar examples in SpillerTest.cpp? Thanks!
```
  explicit SpillerTest(const TestParam& param)
      : param_(param),
        type_(param.type),
        executorPoolSize_(param.poolSize),
        compressionKind_(param.compressionKind),
        hashBits_(0, type_ == Spiller::Type::kOrderBy ? 0 : 2),
        numPartitions_(hashBits_.numPartitions()),
        statWriter_(std::make_unique<TestRuntimeStatWriter>(stats_)) {
    setThreadLocalRunTimeStatWriter(statWriter_.get());
  }

  ~SpillerTest() {
    setThreadLocalRunTimeStatWriter(nullptr);
  }
```